doc/pac/Port Access Control.md

```diff
@@ -0,0 +1,852 @@
+3. On successful authentication of a client, hostapd sends a "Client Authenticated" message to pacd with all the authorization parameters like VLAN, Session-Timeout, etc.
+4. pacd proceeds to authorize the client. RADIUS authorization parameters like client VLAN membership, is communicated to relevant modules (VLAN, FDB) by writing on their tables on STATE_DB.  Authenticated clients are updated in PAC_AUTHENTICATED_CLIENT_OPER table in STATE_DB.
+5. VLAN, FDB further process these STATE_DB updates from PAC and write into their STATE_DB and APPL_DB tables.
+6. Orchagent in SWSS docker gets notified about changes in APPL_DB and responds by translating the APPL_DB changes to respective sairedis calls.
+7. Sairedis APIs write into ASIC_DB.
+8. Syncd gets notified of changes to ASIC_DB and in turn calls respective SAI calls. The SAI calls translate to respective SDK calls to program hardware.
+9. EAP Success message (EAPoL PDU) is sent to the client.
+
+
+### 3.1.3 MAB packet receive flow
+
+![pac-mab-rx-flow](images/PAC_MAB_Rx_Flow.JPG)
+
+
+**Figure 4: MAB PDU receive flow**
+
+1. Unknown source MAC packets are received by hardware on a front panel interface and trapped to CPU. The packets gets delivered to a pacd socket.
+2. pacd sends a "Client Authenticate" message along with the received packet MAC to mabd.
+3. mabd interacts with RADIUS server to authenticate the given client based on the MAC.
+4. On successful authentication of a client, mabd sends an "Client Authenticated" message to pacd with all the authorization parameters like VLAN, Session-Timeout, etc.
+5. pacd proceeds to authorize the client. RADIUS authorization parameters like client VLAN membership, is communicated to relevant modules (VLAN, FDB) by writing on their tables on STATE_DB.  Authenticated clients are updated in PAC_AUTHENTICATED_CLIENT_OPER table in STATE_DB.
+6. VLAN, FDB further process these STATE_DB updates from PAC and write into their STATE_DB and APPL_DB tables.
+7. Orchagent in SWSS docker gets notified about changes in APPL_DB and responds by translating the APPL_DB changes to respective sairedis calls.
+8. Sairedis APIs write into ASIC_DB.
+9. Syncd gets notified of changes to ASIC_DB and in turn calls respective SAI calls. The SAI calls translate to respective SDK calls to program hardware.
+10. EAP success message (EAPoL PDU) is sent to the client.
+
+
+### 3.1.4 RADIUS
+
+PAC uses the RADIUS client from hostapd. 
+
+PAC supports only 1 RADIUS server. The highest priority server will be picked up for authentication. 
+
+
+## 3.2 DB Changes
+
+### 3.2.1 Config DB
+
+**PAC_PORT_CONFIG**   
+```   
+"PAC_PORT_CONFIG": {
+  "Ethernet1": {
+    "method_list": [
+      "dot1x",
+      "mab"
+    ],
+    "priority_list": [
+      "dot1x",
+      "mab"
+    ],
+    "port_pae_role": "authenticator",
+    "port_control_mode": "auto",
+    "host_control_mode": "multi_auth",
+    "reauth_period": 60,
+    "reauth_enable": "true",
```

Comment:
Just a suggestion, maybe consider removing this field and indicting disabled by setting reauth_period as 0. This would be inline with how its done in hostapd.conf 

# EAP reauthentication period in seconds (default: 3600 seconds; 0 = disable
# reauthentication).
#eap_reauth_period=3600